### PR TITLE
Move persist tracking context out of the loop in AbstractFlushingEventListener

### LINKED ISSUE #3307: IsDirty performance hit from 5.3.16 to 5.4.2
I faced a longer IsDirty run after upgrading.

NHibernate 3.3.1:

image

NHibernate 5.4.2:

image

The first question is, was there a bug in old implementation that now lead to this cascade?
Since the result of IsDirty is true in both cases, I can imagine that there is optimization potential.

The test case of IsDirty is there that we have a table which stores for every element a parent element (not lazy load). The children are as HasMany relation. We remove just near all elements, and this lead to this bad result.

### PR DESCRIPTION
**Pull request by hazzik**
Fixes #3307

### CODE REVIEW

File: src/NHibernate/Event/Default/AbstractFlushingEventListener.cs [hunk 1]
```diff
@@ -196,7 +197,7 @@ protected virtual void PrepareEntityFlushes(IEventSource session)
 				Status status = entry.Status;
 				if (status == Status.Loaded || status == Status.Saving || status == Status.ReadOnly)
 				{
-					CascadeOnFlush(session, entry.Persister, me.Key, Anything);
+					CascadeOnFlush(session, entry.Persister, me.Key, anything);
```
Comment (fredericDelaporte) [inline src/NHibernate/Event/Default/AbstractFlushingEventListener.cs:200]: I do not get it. The property implementation just always directly yield null, and is not overridden anywhere to do anything else. What does this change?

Comment (bahusoid) [reply src/NHibernate/Event/Default/AbstractFlushingEventListener.cs:200]: > not overridden anywhere

Not until #2752. Looks like you check on 5.3 codebase.

Comment (hazzik) [reply src/NHibernate/Event/Default/AbstractFlushingEventListener.cs:200]: Yes, as Roman said: it was overridden in #2752:

https://github.com/nhibernate/nhibernate-core/blob/170210aa211a522f145c61437d4429af794f1f7f/src/NHibernate/Event/Default/DefaultDirtyCheckEventListener.cs#L16

See the discussion starting in https://github.com/nhibernate/nhibernate-core/issues/3307#issuecomment-1562296466 for more context.

Comment (fredericDelaporte) [reply src/NHibernate/Event/Default/AbstractFlushingEventListener.cs:200]: Indeed, I was not on the adequate code base.

**Review by fredericDelaporte — APPROVED** [review]

**Pull request merged**